app/models/decision.rb

```diff
@@ -9,7 +9,22 @@ class Decision < ApplicationRecord
   validates :status, inclusion: { in: ["granted", "refused"],
     message: "Please select Yes or No" }
 
+  validate :ensure_correction, on: :update
+
   def comment_made?
     granted? && comment_met.present? || refused? && comment_unmet.present?
   end
+
+  private
+
+  def ensure_correction
+    return unless planning_application.awaiting_determination?
+    return unless user.reviewer?
+    return if planning_application.reviewer_decision.status ==
+        planning_application.assessor_decision.status
```

Comment:
can we extract them into separate methods please?

Reply:
I've added a method on the planning application https://github.com/unboxed/bops/pull/100/commits/6e57d9b41f1371acaf5cdbda665ee72cc1e4f174